workflow/util/retry.go

```diff
@@ -0,0 +1,110 @@
+package util
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+)
+
+// GetFailHosts returns slice of all child nodes with fail or error status
+func GetFailHosts(nodes wfv1.Nodes, parent string) []string {
```

Comment:
maybe we could move this all to it's own package? workflow/util/retry?

Reply:
As you suggested, I moved `retry.go` and `retry_test.go` to `workflow/util/retry` directory.